core/src/main/java/com/linecorp/armeria/common/HttpHeaderGetters.java

```diff
@@ -0,0 +1,310 @@
+     *         no such header or it can't be converted to milliseconds.
+     */
+    long getTimeMillis(CharSequence name, long defaultValue);
+
+    /**
+     * Returns {@code true} if a header with the {@code name} exists, {@code false} otherwise.
+     *
+     * @param name the header name
+     */
+    boolean contains(CharSequence name);
+
+    /**
+     * Returns {@code true} if a header with the {@code name} and {@code value} exists.
+     *
+     * @param name the header name
+     * @param value the header value of the header to find
+     */
+    boolean contains(CharSequence name, CharSequence value);
+
+    /**
+     * Returns {@code true} if a header with the {@code name} and {@code value} exists.
+     *
+     * @param name the header name
+     * @param value the header value
+     * @return {@code true} if the header exists. {@code false} otherwise
+     */
+    boolean containsObject(CharSequence name, Object value);
+
+    /**
+     * Returns {@code true} if a header with the {@code name} and {@code value} exists.
+     *
+     * @param name the header name
+     * @param value the header value
+     * @return {@code true} if the header exists. {@code false} otherwise
+     */
+    boolean containsInt(CharSequence name, int value);
+
+    /**
+     * Returns {@code true} if a header with the {@code name} and {@code value} exists.
+     *
+     * @param name the header name
+     * @param value the header value
+     * @return {@code true} if the header exists. {@code false} otherwise
+     */
+    boolean containsLong(CharSequence name, long value);
+
+    /**
+     * Returns {@code true} if a header with the {@code name} and {@code value} exists.
+     *
+     * @param name the header name
+     * @param value the header value
+     * @return {@code true} if the header exists. {@code false} otherwise
+     */
+    boolean containsFloat(CharSequence name, float value);
+
+    /**
+     * Returns {@code true} if a header with the {@code name} and {@code value} exists.
+     *
+     * @param name the header name
+     * @param value the header value
+     * @return {@code true} if the header exists. {@code false} otherwise
+     */
+    boolean containsDouble(CharSequence name, double value);
+
+    /**
+     * Returns {@code true} if a header with the {@code name} and {@code value} exists.
+     *
+     * @param name the header name
+     * @param value the header value
+     * @return {@code true} if the header exists. {@code false} otherwise
+     */
+    boolean containsTimeMillis(CharSequence name, long value);
+
+    /**
+     * Returns the number of headers.
+     */
+    int size();
+
+    /**
+     * Returns {@code true} if this headers does not contain any entries.
+     */
+    boolean isEmpty();
+
+    /**
+     * Returns a {@link Set} of all header names. The returned {@link Set} cannot be modified.
+     */
+    Set<AsciiString> names();
```

Comment:
Should the iteration methods return `CharSequence` instead of `AsciiString`? All the other methods take `CharSequence` as the type of the key.

Reply:
It is merely to accept both `AsciiString` and `String`. A header name is internally converted to `AsciiString` using `HttpHeaderNames.of()`. So, here, we accept the widest type and expose the narrowest type. I think a user will expect `AsciiString` or `String` rather than `CharSequence` because it lacks useful methods like comparison.